workload/schemachange: add multi-region zone config validation

otan opened the pull request:
Perform validation on multi-region zone configurations using the
built-in. The built-in will return an error if there is a violation.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62933)
<!-- Reviewable:end -->

otan committed:
workload/schemachange: add multi-region zone config validation

Perform validation on multi-region zone configurations using the
built-in. The built-in will return an error if there is a violation.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62933#-:-MXAqmhcAc2O12q0RwKr:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajwerner
